src/main/scala/org/broadinstitute/dsde/firecloud/service/ExportEntitiesByTypeActor.scala

```diff
@@ -369,4 +384,81 @@ class ExportEntitiesByTypeActor(rawlsDAO: RawlsDAO,
       response.results
     }
 
+  /**
+    * Perform file-matching for this workspace's bucket. Lists the files in the bucket, filtered by a bucket prefix,
+    * then executes `FileMatcher.pairPaths` on those files. Finally, creates a TSV out of the paired results.
+    *
+    * @see [[FileMatcher]]
+    * @param matchingOptions configuration options for file matching
+    * @return contents of the resultant TSV
+    */
+  def matchBucketFiles(matchingOptions: FileMatchingOptions): Future[String] = {
+    // generate defaults for options
+    val read1Name = matchingOptions.read1Name.getOrElse("read1")
+    val read2Name = matchingOptions.read2Name.getOrElse("read2")
+    val recursive = matchingOptions.recursive.getOrElse(true)
+
+    validateUserDefinedString(read1Name)
+    validateUserDefinedString(read2Name)
+    validateAttributeName(AttributeName.fromDelimitedName(read1Name), entityType)
+    validateAttributeName(AttributeName.fromDelimitedName(read2Name), entityType)
+
+    // retrieve workspace so we can get its bucket
+    rawlsDAO.getWorkspace(workspaceNamespace, workspaceName)(userInfo) map { workspaceResponse =>
+      val workspaceBucket = GcsBucketName(workspaceResponse.workspace.bucketName)
+
+      // list all files in bucket which match matchingOptions.prefix
+      logger.info("listing bucket files ...")
+      val fileList: List[GcsObjectName] =
+        googleServicesDao.listBucket(workspaceBucket, Option(matchingOptions.prefix), recursive)
+
+      logger.info(s"found ${fileList.length} files")
```

Comment:
do we want to place a limit on this based on the number of files returned?

Reply:
added in 9505b6e09871ef060f002b74a893e463a1ae54ec